docs/source/extending/frontend_extensions.rst

```diff
@@ -5,125 +5,210 @@ This describes the basic steps to write a JavaScript extension for the Jupyter
+again this is an API which is not guaranteed to be stable. However, custom key
+bindings are frequently requested, and are helpful to increase accessibility,
+so in the following we show how to access them.
 
-Here is an example of extension that will unbind `0,0` in command mode, which
-normally restarts the kernel, and bind `0,0,0` in it's place.
+Here is an example of an extension that will unbind the shortcut ``0,0`` in
+command mode, which normally restarts the kernel, and bind ``0,0,0`` in its
+place:
 
 .. code:: javascript
 
-    //file myext/main.js
+    // file my_extension/main.js
 
-    define(['base/js/namespace'],function(Jupyter){
+    define([
+        'base/js/namespace'
+    ], function(
+        Jupyter
+    ) {
 
-        function load_ipython_extension(){
-            Jupyter.keyboard_manager.command_shortcuts.remove_shortcut('0,0')
-            Jupyter.keyboard_manager.command_shortcuts.add_shortcut('0,0,0', 'jupyter-notebook:restart-kernel')
+        function load_ipython_extension() {
+            Jupyter.keyboard_manager.command_shortcuts.remove_shortcut('0,0');
+            Jupyter.keyboard_manager.command_shortcuts.add_shortcut('0,0,0', 'jupyter-notebook:restart-kernel');
         }
 
-        return {load_ipython_extension: load_ipython_extension };
-    })
+        return {
+            load_ipython_extension: load_ipython_extension
+        };
+    });
 
 .. note::
     
-    Keybindings might not work correctly on non-US keyboards.
-    Unfortunately, this is a limitation of browser implementations and the status of
-    keyboard event handling on the web. We appreciate your feedback if you have
-    issues binding keys, or ideas to help improve the situation.
-
-You can see that I have used the **command name**
-`jupyter-notebook:restart-kernel` to bind the new shortcut. There is no API yet
-to access the list of all available *commands*, though the following in the
+    The standard keybindings might not work correctly on non-US keyboards.
+    Unfortunately, this is a limitation of browser implementations and the
+    status of keyboard event handling on the web in general. We appreciate your
+    feedback if you have issues binding keys, or have any ideas to help improve
+    the situation.
+
+You can see that I have used the **action name**
+``jupyter-notebook:restart-kernel`` to bind the new shortcut. There is no API
+yet to access the list of all available *actions*, though the following in the
 JavaScript console of your browser on a notebook page should give you an idea
-of what of available:
+of what is available:
 
 .. code:: javascript
 
-    Object.keys(IPython.actions._actions)
+    Object.keys(require('base/js/namespace').actions._actions);
 
-In this example, we changed keyboard shortcuts in **command mode**; you can also
-customise keyboard shortcuts in edit mode.
+In this example, we changed the keyboard shortcut in the **command mode**; you
```

Comment:
'...changed *a* keyboard shortcut in command mode' (both instances of 'the' read wrong here)